docs/semgrep-supply-chain/dependency-search.md

```diff
@@ -72,6 +72,36 @@ Dependency search provides the following filters, which correspond to the data p
 ![Screenshot of dependency search with query](/img/SSC-DepSearch-Query.png#md-width)
 _**Figure**. Dependency search page with sample search query._
 
+## Dependency graphs (beta)
+
+:::info
+To participate in this beta, reach out to [sales@semgrep.com](mailto:sales@semgrep.com).
+:::
+
+Dependency graphs allow you to view dependency paths for all transitive dependencies, up to seven layers deep, introduced in a project. With this information, you can understand:
+
+- How a transitive dependency was introduced
+- How deep the transitive dependency is nested in the dependency tree. The dependency tree reflects your project that Semgrep generates
+
+Semgrep is capable of generating dependency graphs for:
+
+- Java projects with lockfiles and Java projects **without lockfiles** if they're built using Maven or Gradle with the help of the Gradle Wrapper. Ensure that the environment where you run Semgrep has installed the necessary dependencies, such as Java and Maven, that are required to build your project
+- Java projects that include a `maven_dep_tree.txt` file.
+
+Dependency graphs are available once you've updated your Semgrep deployment to use the `--allow-local-builds` flag when initiating the scan from an environment with all of the project's required dependencies installed:
+
+```console
+semgrep ci --allow-local-builds
+```
```

Comment:
If the user has `maven_dep_tree.txt`, it is not necessary for them to use this flag; a normal Semgrep scan will work